compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5135,6 +5135,8 @@ private void visitCheckAndCheckPanicExpr(BLangCheckedExpr checkedExpr) {
         BType typeOfExprWithCheckingKeyword;
         if (expType == symTable.noType) {
             typeOfExprWithCheckingKeyword = symTable.noType;
+        } else if (expType.tag == TypeTags.UNION && types.getErrorType((BUnionType) expType) != null) {
+            typeOfExprWithCheckingKeyword = expType;
```

Comment:
```ballerina
type Err distinct error;

type ErrTwo distinct error;

public function main() returns Err? {
    int|ErrTwo x = check bar();
}

function bar() returns int|error {
    return error("");
}
```

This now results in the following error, which isn't quite correct, right?
```cmd
incompatible types: expected '(int|ErrTwo)', found '(int|error)'
```

Previously the error was 
```cmd
invalid usage of the 'check' expression operator: no matching error return type(s) in the enclosing invokable
```

Reply:
This is fixed now. Thank you.